### PR TITLE
[#172] Service Selector refactor

### PR DESCRIPTION
**Pull request by wujuu**
Added softmax in the end - needed for proper masking

### CODE REVIEW

File: recommender/engine/agents/rl_agent/service_selector.py [hunk 1]
```diff
@@ -54,7 +51,8 @@ def __call__(
         if (mask > 0).sum() < K:
             raise InsufficientRecommendationSpace
 
-        engagement_values = weights @ self.itemspace.T
+        engagement_values = F.softmax(weights @ self.itemspace.T)
```
[the remark this answers is not in the document]
Comment (wujuu) [reply recommender/engine/agents/rl_agent/service_selector.py:54]: It does work with softmax though. It is needed for proper masking as stated in the PR description
